index.html

```diff
@@ -833,23 +839,70 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
 'PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer'>performance
 entry buffer</a>.</li>
 <li>Increase <a>resource timing buffer current size</a> by 1.</li>
+<li>Return.</li>
 </ol>
 </li>
-<li>If <a>resource timing buffer current size</a> is greater than
-or equal to <a>resource timing buffer size limit</a> and the
-<a>resource timing buffer full flag</a> is false, run the following
+<li>If <a>resource timing buffer full flag</a> is false, run the following
 substeps:
-<ol style='list-style-type: lower-latin;' data-link-for=
-"Performance">
-<li>Set the <a>resource timing buffer full flag</a> to true.</li>
-<li><a data-cite="DOM#concept-event-fire">Fire an event</a> named
-<code>resourcetimingbufferfull</code> at the <a data-cite=
-"HR-TIME-2#idl-def-performance">Performance</a> object, with its
-<code>bubbles</code> attribute initialized to true.</li>
+<ol style="list-style-type: lower-latin;">
+<li>Set <a>resource timing buffer full flag</a> to true.</li>
+<li>Queue a task to run <a>fire a buffer full event</a>.</li>
+</ol>
+<li>Add <i>new entry</i> to the <a>secondary entry buffer</a>.</li>
+</ol>
+
+<p>To <dfn>copy secondary buffer</dfn>, run the following steps:</p>
+<ol>
+<li>While <a>secondary entry buffer</a> is not empty, run the following substeps:
+<ol>
+<li>Let <i>entry</i> be the oldest <a>PerformanceResourceTiming</a> in
+<a>secondary entry buffer</a>.</li>
+<li>Add <i>entry</i> to the end of <a data-cite=
+"PERFORMANCE-TIMELINE-2/#dfn-performance-entry-buffer">performance entry
+buffer</a>.</li>
+<li>Increment <a>resource timing buffer current size</a> by 1.</li>
+<li>Remove <i>entry</i> from <a>secondary entry buffer</a>.</li>
+</ol>
+<li>Assert that <a>secondary entry buffer</a> is empty.</li>
+</ol>
+
+<p>To <dfn>fire a buffer full event</dfn>, run the following steps:</p>
+<ol>
+<li><a data-cite="DOM/#concept-event-fire">Fire an event</a>
+named <code>resourcetimingbufferfull</code> at the <a data-cite=
+"hr-time-2/#idl-def-performance">Performance</a> object,</li>
+<li>Run <a>copy secondary buffer</a>.</li>
+<li>While <a>resource timing buffer current size</a> is larger than 
```

Comment:
And then this while asks about emptiness of secondary buffer instead of primary buffer size.

Reply:
yeah